src/main/java/com/google/devtools/build/lib/rules/repository/RepositoryDelegatorFunction.java

```diff
@@ -685,39 +685,38 @@ byte[] areRepositoryAndMarkerFileConsistent(
       String content;
       try {
         content = FileSystemUtils.readContent(markerPath, UTF_8);
-        String markerRuleKey = readMarkerFile(content, recordedInputValues);
-        boolean verified = false;
-        if (Preconditions.checkNotNull(ruleKey).equals(markerRuleKey)) {
-          verified = handler.verifyRecordedInputs(rule, directories, recordedInputValues, env);
-          if (env.valuesMissing()) {
-            return null;
-          }
+        if (!readMarkerFile(content, Preconditions.checkNotNull(ruleKey), recordedInputValues)) {
+          return null;
         }
-
-        if (verified) {
-          return new Fingerprint().addString(content).digestAndReset();
-        } else {
+        if (!handler.verifyRecordedInputs(rule, directories, recordedInputValues, env)) {
           return null;
         }
+        if (env.valuesMissing()) {
+          return null;
+        }
+        return new Fingerprint().addString(content).digestAndReset();
       } catch (IOException e) {
         throw new RepositoryFunctionException(e, Transience.TRANSIENT);
       }
     }
 
     @Nullable
-    private static String readMarkerFile(
-        String content, Map<RepoRecordedInput, String> recordedInputValues) {
-      String markerRuleKey = null;
+    private static boolean readMarkerFile(
```

Comment:
we can probably make this even cleaner by returning a `Map<RepoRecordedInput, String>` instead. When you'd otherwise return `false`, simply return an `ImmutableMap.of(RepoRecordedInput.NEVER_UP_TO_DATE, "")`.

Reply:
That's a good point. In the new code, I could also replace `if (!firstLineVerified)` with `recordedInputValues == null` or even `recordedInputValues == notUpToDate`, but I though the boolean makes the logic more clear.